students/TrkCat/Lesson04/mailroom.py

```diff
@@ -0,0 +1,141 @@
+    donate_menu = {'1': add_donation,
+                   '2': list_donors,
+                   '3': quit_menu
+                   }
+    show_menu(donate_prompt, donate_menu)
+
+
+def create_report():
+    """Print a donation report"""
+    sorted_donors = sorted(donors.values(), key=lambda v: v['tot_don'],
+                           reverse=True)
+    # Add in dynamic column widths in future iteration
+    header = ' | '.join(('     Donor Name    ', 'Total Given', 'Num Gifts',
+                         'Average Gift'))
+    print('\n' + header)
+    print('-' * len(header))
+    for donor in sorted_donors:
+        print(('{name:20s}  $ {tot_don:>10.2f}   {num_don:>9d}'
+               '  $ {avg_don:>10.2f}').format(**donor))
+
+
+def send_all_letters():
+    write_dir = input('\nWhich directory should the letters be writen in?: ')
+    if not write_dir:
+        write_dir = '.'
+    for donor in donors.values():
+        letter = gen_email()
+        file_name = donor['name'] + '.txt'
+        print('Writing Letter: {}'.format(file_name))
+        with open('/'.join([write_dir, file_name]), 'w') as f:
+            f.write(letter.format(**donor))
+    print('\nAll letters complete.\n')
+
+
+def add_donation():
+    """Add donation amount to list under donors name"""
+    donor_name = input("\nPlease enter donor's full name: ")
+    if donor_name.lower() not in donors:
+        donors[donor_name.lower()] = {'name': donor_name}
+    cur_dict = donors[donor_name.lower()]
+    donation = float(input('Enter the donation amount: '))
+    cur_dict.setdefault('donations', []).append(donation)
+    cur_dict['last_don'] = donation
+    update_tot_avg(cur_dict)
+    letter = gen_email()
+    print(letter.format(**cur_dict))
+    return False
+
+
+def list_donors():
+    """Print list of current donors"""
+    print('\n\nCurrent Donors:\n')
+    for donor in donors.values():
+        print(donor['name'])
+
+
+def gen_email():
+    """Return a thank you email"""
+    letter = ('\nFROM: Your friendly local charity mailroom.\n'
+              'TO: {name}\n'
+              'RE: Your recent donation\n\n'
+              '\nThank you so much for your recent donation of'
+              ' ${last_don:,.2f}. This will go a long way towards helping to'
+              ' save the pythons. Your generosity is most appreciated!'
+              '\n\nBest Regards,\nSave The Pythons\n'
+              )
+    return letter
+
+
+def quit_menu():
+    """Quit current menu"""
+    return False
+
+
+def update_tot_avg(don_dict):
+    """Update total and average donation value in dict d."""
+    don_dict['tot_don'] = sum(don_dict['donations'])
+    don_dict['num_don'] = len(don_dict['donations'])
+    don_dict['avg_don'] = don_dict['tot_don']/don_dict['num_don']
+
+
+def show_menu(prompt, disp_dict):
+    """Generate menu with dispatch dictionary"""
+    while True:
+        sel = input(prompt)
+        if disp_dict[sel]() is False:
+            return False
```

Comment:
since you're not consuming return value from this function, you can just do `return` or `break`